### DISCUSSION TITLE
Stripe works in emulator but does not work in Test Flight

### DISCUSSION THREAD

**Discussion by MartinBal97** [flutter-stripe/flutter_stripe#1718] [Q&A]
**Describe the bug**
Stripe modal close in test flight 2 seconds after was open, but in my emulator is working perfect.

**To Reproduce**
1. Just call the funtion that opens the stripe modal in test flight

**Expected behavior**
It should let me add my card number etc and not to close by itself

**Smartphone / tablet**
 - Device: [Every iPhone by the moment] NOT emulator included

**Additional context**
This video show exacly what is the error.

https://github.com/flutter-stripe/flutter_stripe/assets/47724887/83756140-4052-435d-ae63-36252e5421d6

**Comment by remonh87** [comment]
Hi this normally happens when there is something configured wrong in your app. Did you for example enable apple pay?  Also if you do not use testmode anymore you have to use the stripe production key and it should be configured. 

I would recommend running the app on your phone through xcode and send me the logs for further investigation.

**Reply by MartinBal97** [reply]
Hi! Thank you for your response.  @remonh87 

In my flavors, the app that is in TestFlight is using the public key (Stripe dev key), and Apple Pay and Google Pay are not enabled at the moment. The app that is in production is using the Stripe production key. The Stripe production key (private key) and the Stripe dev key (public key) are configured and working on the web app (which wasn't made with Flutter), but in the mobile app, Stripe is failing in test mode. I don't know how it's working in production.

I just ran the app on my phone through Xcode, and it's working fine. So I can't get the logs because it's only failing on TestFlight.

**Reply by MartinBal97** [reply]
Hi, this is the log, the deploy is with codemagic, and the environment variables are configured correctly. @remonh87 

StripeException(error:
LocalizedErrorMessage code:
FailureCode.Failed, localizedMessage:
Se produjo un error inesperado. Vuelve a intentar en unos segundos., message:
You did not provide an API key. You need to provide your API key in the Authorization header, using Bearer auth (e.g. Authorization: Bearer
YOUR_SECRET_KEY'). See https:// stripe.com/docs/api#authentication for details, or we can help at https:// support.stripe.com/., stripeErrorCode:, declineCode: null, type: invalid_request_error))

**Comment by Hiti3** [comment]
Anyone solved this???

**Reply by MartinBal97** [reply]
Hello Hiti3, yes I did, long time ago, I have to remember how haha, but it was a flavors issue, do you have flavors in your app?
